### PR TITLE
METAL-3280 Add a job for checking chart with different values

### PR DESCRIPTION
**Pull request by allanger**
I've added a couple of steps to the "Release Charts" workflow.

There is a tool [chart-testing](https://github.com/helm/chart-testing). I've had a look at Prometheus helm chart and found out that they are using it to lint charts, so I've decided to try it. There is a [github action](https://github.com/helm/chart-testing-action) for it already. 

What have I done:

- Added this step to setup Python which is required for ct to work

```
      - uses: actions/setup-python@v2
        with:
          python-version: 3.7
```

- Added this step to setup chart-testing

```
      - name: Set up chart-testing
        uses: helm/chart-testing-action@v2.2.0
```

- And added step for linting the chart
```
      - name: Run chart-testing (lint)
        run: ct lint --validate-maintainers=false --check-version-increment=false
```

Flags explanation:
```
--validate-maintainers=false 
```
By default `ct` requires that maintainers are specified in chart. Maybe we should add this information in the `Chart.yaml` and remove the flag? 

```
--check-version-increment=false
```
If enabled, version increment check is a blocking thing.  As for me, it doesn't make a sense to check the version update on every push to master if the chart is in the same repo as the code. Because the chart may not be updated every time the code is.  

Here is an example of this workflow: https://github.com/kloeckner-i/db-operator/runs/4763174508?check_suite_focus=true 
I've turned off all conditions to check how it's going.

Also I've added several `*-values.yaml` files to `/charts/db-operator/ci` that are used by `ct` to run linter.

### CODE REVIEW

File: .github/workflows/chart-release.yaml [hunk 1]
```diff
@@ -23,11 +22,21 @@ jobs:
...
         uses: azure/setup-helm@v1
         with:
           version: v3.4.1
-          
+
+      - name: Set up chart-testing
+        uses: helm/chart-testing-action@v2.2.0
+
+      - name: Run chart-testing (lint)
+        run: ct lint --validate-maintainers=false --check-version-increment=false
```
Comment (dabde) [inline .github/workflows/chart-release.yaml:38]: think the ct lint should be part of all PR pipelines. So `build-and-test.yaml` with the normal lint job. or you would only see a wrong value constellation after it's already in master.

**Review by dabde — CHANGES_REQUESTED** [review]
Move the job to be run every time in a branch.

File: .github/workflows/chart-release.yaml (hunk 1 above)
Comment (allanger) [reply .github/workflows/chart-release.yaml]: I've removed this job from chart-release pipeline and create `helm-lint` job in the `build-and-test` one. 

Now it's a separate job. Example: https://github.com/kloeckner-i/db-operator/actions/runs/

There are still a couple of changes in [chart-release.yaml](https://github.com/kloeckner-i/db-operator/blob/158c452dbce3b8ba3b65c0c7d8f8ab333154e86e/.github/workflows/chart-release.yaml) but those are formatting things.

File: charts/db-operator/ci/ci-2-values.yaml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  backup:
+    activeDeadlineSeconds: 600  # 10m
+    nodeSelector:
+      nodetype: database
+    postgres:
+      image: "kloeckneri/pgdump-gcs:latest"
+    mysql:
+      image: "kloeckneri/mydump-gcs:latest"
+  monitoring:
+    promPushGateway: ""
```
Comment (dabde) [inline charts/db-operator/ci/ci-2-values.yaml:58]: always empty. would not test this setting.

File: charts/db-operator/ci/ci-2-values.yaml [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+relabelings:
+  - sourceLabels: [__meta_kubernetes_pod_node_name]
+    separator: ;
+    regex: ^(.*)$
+    targetLabel: nodename
+    replacement: $1
+    action: replace
+
+service:
+  annotations: {}
```
Comment (dabde) [inline charts/db-operator/ci/ci-2-values.yaml:122]: also one case with value missing.

File: charts/db-operator/ci/ci-2-values.yaml [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  - key: "key2"
+    operator: "Equal"
+    value: "value2"
+    effect: "NoSchedule"
+
+config:
+  instance:
+    google:
+      proxy:
+        nodeSelector: {}
```
Comment (dabde) [inline charts/db-operator/ci/ci-2-values.yaml:41]: also one case with value missing.

**Review by dabde — CHANGES_REQUESTED** [review]
some cases are not tested. everything else is fine.

**Review by dabde — APPROVED** [review]

File: charts/db-operator/ci/ci-2-values.yaml (hunk 2 above)
Comment (allanger) [reply charts/db-operator/ci/ci-2-values.yaml]: Added this value to ci-2-values and ci-3-values
```
promPushGateway: prometheus-pushgateway.monitoring:9090
```

**Pull request merged**